tensorflow_addons/rnn/layernorm_simplernn.py

```diff
@@ -0,0 +1,434 @@
+      kernel_constraint: Constraint function applied to the `kernel` weights
+        matrix. Default: `None`.
+      recurrent_constraint: Constraint function applied to the `recurrent_kernel`
+        weights matrix. Default: `None`.
+      bias_constraint: Constraint function applied to the bias vector
+         (`use_bias=True`) or for the beta vector of the layer normalization
+         layer (`use_layernorm=True`). Default: `None`.
+      gamma_constraint: Constraint function applied to the gamma vector
+         of the layer normalization layer (`use_layernorm=True`).
+         Default: `None`.
+      dropout: Float between 0 and 1. Fraction of the units to drop for the
+        linear transformation of the inputs. Default: 0.
+      recurrent_dropout: Float between 0 and 1. Fraction of the units to drop for
+        the linear transformation of the recurrent state. Default: 0.
+
+    Call arguments:
+      inputs: A 2D tensor, with shape of `[batch, feature]`.
+      states: A 2D tensor with shape of `[batch, units]`, which is the state
+        from the previous time step. For timestep 0, the initial state provided
+        by the user will be feed to cell.
+      training: Python boolean indicating whether the layer should behave in
+        training mode or in inference mode. Only relevant when `dropout` or
+        `recurrent_dropout` is used.
+
+    Examples:
+
+    ```python
+    inputs = np.random.random([32, 10, 8]).astype(np.float32)
+    rnn = tf.keras.layers.RNN(
+      tf.keras.layers.LayernormSimpleRNNCell(4, use_layernorm=True))
+
+    output = rnn(inputs)  # The output has shape `[32, 4]`.
+
+    rnn = tf.keras.layers.RNN(
+        tf.keras.layers.LayernormSimpleRNNCell(4, use_layernorm=True),
+        return_sequences=True,
+        return_state=True)
+
+    # whole_sequence_output has shape `[32, 10, 4]`.
+    # final_state has shape `[32, 4]`.
+    whole_sequence_output, final_state = rnn(inputs)
+    ```
+    """
+
+    def __init__(
+            self,
+            units,
+            activation='tanh',
+            use_bias=True,
+            use_layernorm=False,  # NEW(!)
+            layernorm_epsilon=1e-05,  # NEW(!)
+            kernel_initializer='glorot_uniform',
+            recurrent_initializer='orthogonal',
+            bias_initializer='zeros',
+            gamma_initializer='ones',  # NEW(!)
+            kernel_regularizer=None,
+            recurrent_regularizer=None,
+            bias_regularizer=None,
+            gamma_regularizer=None,  # NEW(!)
+            kernel_constraint=None,
+            recurrent_constraint=None,
+            bias_constraint=None,
+            gamma_constraint=None,  # NEW(!)
+            dropout=0.,
+            recurrent_dropout=0.,
+            **kwargs):
+        self.use_layernorm = use_layernorm
+        SimpleRNNCell.__init__(
+            self,
+            units,
+            activation=activation,
+            use_bias=False if use_layernorm else use_bias,
```

Comment:
you would add a constant term twice. The `beta` term in `LayerNormalization` is a bias term too, and replaces the normal `bias` term. The authors also don't add a classic bias before scaling in layer normalization, they add the bias term `beta` afterwards.

here is an explanation 
https://github.com/tensorflow/tensorflow/pull/35469#issuecomment-569532428

Ah. Yes. We can do something else. We can deactivate the beta (bias) term in `LayerNormalization(center=False,...)` (Only do scaling) and apply `self.bias` afterwards. This would remove this weird `False if use_layernorm else use_bias` and some corrections in `get_config`

Reply:
`centering` in `self.layernorm` is now deactivated. I added a docstring with formulas as explanation how layer normalization (only scaling) works together with bias, dropout, and activation. https://github.com/tensorflow/addons/pull/841/commits/1d320413498cd33efc7eefa2b51bb2e78f87ef05

Reply:
`use_layernorm` is gone